packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -404,20 +406,39 @@ class TransactionService {
       transaction._start = earliestSpan._start
     }
 
-    /**
-     * Adjust end time of the transaction to match the latest
-     * span end time
-     */
-    const latestSpan = getLatestNonXHRSpan(spans)
-    if (latestSpan && latestSpan._end > transaction._end) {
-      transaction._end = latestSpan._end
+    const latestSpan = getLatestNonXHRSpan(spans) || {}
+    const latestSpanEnd = latestSpan._end || 0
+
+    // Before ending the page-load transaction we are adding a delay to monitor events such as LCP and network requests.
+    // We need to make sure that we are not adding that extra time to the transaction end time
+    // if nothing has been monitored or if the last monitored event end time is less than the delay.
+    if (transaction.type === PAGE_LOAD) {
+      const transactionEndWithoutDelay = transaction._end - PAGE_LOAD_DELAY
+      const lcp = metrics.lcp || 0
+      const latestXHRSpan = getLatestXHRSpan(spans) || {}
```

Comment:
Do we need this still? I forgot why we did added this check
1. Since the transaction always waits for the XHR tasks which are blocking, we are covered in this scenario which means transaction.end will always be greater than XHR span end. 

Let me know if I got it wrong.

Reply:
This check is making sure that the XHR tasks with the following characteristics are handled:

1. triggered during the page load delay
2. finished **before** the delay

Let's imagine a scenario with this timeline

1. Non XHR span               90ms
2. LCP entry                      120 ms
3. page fires load event   135ms
4. 1 second delay is set

During that 1-second window where we observe for activity, there are two possible scenarios related to XHR spans that need to be handled:

<ins>**Scenario A**</ins> (the one that you are commenting, where the extra check it's not required)

A request that will last 1200ms have been observed. The timeline then it would be:

5. delay finishes.            1135ms
6. request finishes.        1335ms (and here the transaction ends, with the proper duration as you mentioned)

<img width="394" alt="Screenshot 2022-04-20 at 11 14 40" src="https://user-images.githubusercontent.com/15065076/164194625-e688afc9-ff00-4a13-8730-daf3f240d7ca.png">



<ins>**Scenario B**</ins> (the one that explains the extra check)

A request that will last 300ms have been observed. The timeline then it would be:

5. request finishes.        435ms
6. delay finishes.            1135ms (and here the transaction ends, with the **wrong** duration)

In such scenario we have there a gap of 700ms that we need to adjust, that's why we have that check too, you can see it in the image below

<img width="366" alt="Screenshot 2022-04-20 at 11 15 56" src="https://user-images.githubusercontent.com/15065076/164194895-af4aab36-b294-4bda-8c25-be6e08ef2572.png">

I will add a comment in the codebase explaining the why of this in a concise way

Reply:
Gotcha, that makes sense. Thanks for the details @devcorpio 👍🏽 Lets get this in.